pkg/yttlibrary/overlay/api.go

```diff
@@ -198,6 +206,9 @@ func (b overlayModule) Subset(
 		if _, ok := actualObj.(*yamlmeta.Document); ok {
 			expectedObj = &yamlmeta.Document{Value: expectedObj}
 		}
+		if _, ok := actualObj.(*yamlmeta.MapItem); ok {
```

Comment:
not sure i get why we need this part. shouldnt we be doing something similar to what we do for ArrayItem above?

Reply:
I hoped so, too.

However, when the `Tuple` is exchanged for its node from the AST, it comes back as a `Map` wrapping the `MapItem`, not the `MapItem`, itself.  So, to make the comparison, we need to reach into that `Map`.

In the case where there are no items _in_ the map (e.g. `{}`), we need a zero-value.  I read that as a lower-level detail and hence `itemFromMap()`.

Reply:
Fascinatingly, by passing in the value of menu items, no reshaping was required within the matcher.